components/apimgt/org.wso2.carbon.apimgt.spec.parser/src/main/java/org/wso2/carbon/apimgt/spec/parser/definitions/AbstractAsyncApiParser.java

```diff
@@ -0,0 +1,187 @@
+/*
+ *   Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com)
+ *
+ *   WSO2 LLC. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.spec.parser.definitions;
+
+import org.wso2.carbon.apimgt.api.APIDefinition;
+import org.wso2.carbon.apimgt.api.APIDefinitionValidationResponse;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.api.dto.KeyManagerConfigurationDTO;
+import org.wso2.carbon.apimgt.api.model.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Abstract base class used to create AsyncAPI parsers.
+ * It extends the APIDefinition class and provides common parsing capabilities for AsyncAPI specifications
+ */
+public abstract class AbstractAsyncApiParser extends APIDefinition {
+
+    @Override
+    public Map<String, Object> generateExample(String apiDefinition) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public Set<URITemplate> getURITemplates(String resourceConfigsJSON) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public Set<Scope> getScopes(String resourceConfigsJSON) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String generateAPIDefinition(SwaggerData swaggerData) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public String generateAPIDefinition(SwaggerData swaggerData, String swagger) throws APIManagementException {
+        return null;
+    }
+
+    @Override
+    public APIDefinitionValidationResponse validateAPIDefinition(String apiDefinition, boolean returnJsonContent)
+            throws APIManagementException {
```

Comment:
We dont need to implement this if we are implementing from all the immediate on abstract class. Please check all the other implemented methods here, and remove if such.